codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/Project.kt

```diff
@@ -0,0 +1,143 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.codyze
+
+import com.github.ajalt.clikt.parameters.groups.OptionGroup
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.multiple
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.path
+import de.fraunhofer.aisec.cpg.TranslationConfiguration
+import de.fraunhofer.aisec.cpg.TranslationManager
+import de.fraunhofer.aisec.cpg.TranslationResult
+import io.github.detekt.sarif4k.*
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.Path
+
+/** Options common to all subcommands dealing projects. */
+class ProjectOptions : OptionGroup("Project Options:") {
+    val directory by
+        option("--project-dir", help = "The project directory").path().default(Path("."))
+}
+
+/** Options common to all subcommands dealing with CPG translation. */
+class TranslationOptions : OptionGroup("CPG Translation Options:") {
+    val sources: List<Path>? by
+        option(
+                "--sources",
+                help = "A list of source files. They will be all added to a single component 'app'.",
+            )
+            .path()
+            .multiple()
+    val components: List<String>? by
+        option(
+                "--components",
+                help =
+                    "The components to analyze. They must be located inside the 'components' folder inside the project directory. The 'components' folder will be taken as the topLevel property for the translation configuration.",
+            )
+            .multiple()
+
+    val exclusionPatterns: List<String>? by
+        option("--exclusion-patterns", help = "A pattern of files to exclude").multiple()
+}
+
+/**
+ * Represents the result of the analysis.
+ *
+ * @param translationResult The result of the CPG translation.
+ * @param run The SARIF run object, that contains findings.
+ */
+data class AnalysisResult(val translationResult: TranslationResult, val run: Run)
+
+/** Analyzes the project and returns the result. */
+fun analyze(config: TranslationConfiguration): AnalysisResult {
+    val run =
+        Run(
+            tool = Tool(driver = ToolComponent(name = "Codyze", version = "x.x.x")),
+            results =
+                listOf(
+                    Result(
```

Comment:
This seems a leftover or some sort of preparation for a TODO and should receive a comment